demos/home-automation/ui/components/weatherInfo.slint

```diff
@@ -37,39 +37,34 @@ export component Day inherits Rectangle {
 
         property <length> icon-size: 40px;
 
-        if weather-data.condition == WeatherCondition.sunny: Rectangle {
-            height: 38px;
-            Image {
-                width: AppState.graphics-accelerator-available ? self.source.width * scale * 1px : self.source.width * 1px;
-                source: @image-url("../images/weather/sunny.png");
+        function weather-icon-source(weather-data: WeatherData) -> image {
```

Comment:
I think you can even write this:

```suggestion
        pure function weather-icon-source(weather-data: WeatherData) -> image {
```

Or you could make it just `pure function weather-icon-source() -> image` and use `self.weather-data`.